test/toolkits/test_browser_toolkit.py

```diff
@@ -0,0 +1,198 @@
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+import io
+import os
+from typing import (
+    TYPE_CHECKING,
+)
+
+from PIL import Image
+
+if TYPE_CHECKING:
+    pass
+
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from camel.toolkits.browser_toolkit import (
+    AcceptedBrowserChannels,
+    BaseBrowser,
+    BrowserToolkit,
+)
+
+
+@pytest.fixture(scope="function")
+def base_browser_fixture():
+    with patch('playwright.sync_api.sync_playwright'):
+        browser = BaseBrowser(headless=True, cache_dir="test_cache")
+        yield browser
+        # Cleanup
+        if os.path.exists("test_cache"):
+            os.rmdir("test_cache")
+
+
+@pytest.fixture(scope="function")
+def browser_toolkit_fixture():
+    with patch('playwright.sync_api.sync_playwright'):
+        toolkit = BrowserToolkit(headless=True, cache_dir="test_cache")
+        yield toolkit
+        # Cleanup
+        if os.path.exists("test_cache"):
+            os.rmdir("test_cache")
+
+
+def test_base_browser_init(base_browser_fixture):
+    browser = base_browser_fixture
+    assert browser.headless is True
+    assert browser.cache_dir == "test_cache"
+    assert isinstance(browser.history, list)
+    assert isinstance(browser.page_history, list)
+
+
+def test_base_browser_initialization_order():
+    with patch('playwright.sync_api.sync_playwright'):
+        browser = BaseBrowser(headless=True)  # __init__ called
+        assert not hasattr(browser, 'browser')  # browser not launched yet
+        assert not hasattr(browser, 'page')  # page not created yet
+
+        browser.init()  # explicit init() call
+        assert hasattr(browser, 'browser')  # browser launched
+        assert hasattr(browser, 'page')  # page created
+
+
+def test_browser_chrome_channel_selection():
```

Comment:
I suggest to use `pytest.mark.parametrize` like
https://github.com/camel-ai/camel/blob/9ed1cf74804bc5df8fe6512abe628943464d7be5/test/models/test_vllm_model.py#L25-L33

instead of creating different tests for different channels

Reply:
Fixed.